### PR TITLE
feat(ivy): support undefined and null in static interpreter

### PR DESCRIPTION
**Pull request by JoostK**
Previously, the usage of `null` and `undefined` keywords in code that is
statically interpreted by ngtsc resulted in a `DynamicValue`, as they were
not recognized as special entities. This commit adds support to interpret
these keywords.

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]

**Review by alxhub — APPROVED** [review]

**Comment by JoostK** [comment]
merge-assistance: test_docs_examples failure is a flake & please run presubmit

**Comment by matsko** [comment]
http://test/OCL:258495899:BASE:258495905:1563336002796:a00c6c35

File: packages/compiler-cli/src/ngtsc/partial_evaluator/test/evaluator_spec.ts [hunk 1]
```diff
@@ -172,6 +172,21 @@ runInEachFileSystem(() => {
...
+      expect(evaluate('const a = true;', 'a')).toEqual(true);
+      expect(evaluate('const a = false;', 'a')).toEqual(false);
+    });
+
+    it('supports undefined',
+       () => { expect(evaluate('const a = undefined;', 'a')).toEqual(undefined); });
+
+    it('supports null', () => { expect(evaluate('const a = null;', 'a')).toEqual(null); });
+
+    it('negation works', () => {
```
Comment (JoostK) [inline packages/compiler-cli/src/ngtsc/partial_evaluator/test/evaluator_spec.ts]: Ooops, bad rebase introduced a repeated test. Will remove.

**Comment by JoostK** [comment]
merge-assistance: presubmit was ran yesterday, the additional fixup commit only removes a duplicated test (introduced during rebase)

**Pull request closed without merging**